### PR TITLE
Feat: Image using next/future/image + new Thumbor URL

### PR DESCRIPTION
**Pull request by eduardoformiga**
## What's the purpose of this pull request?

- [x] uses new thumbor domain (vtexassets) - https://www.faststore.dev/releases/2023/03/10/faststore
- [x] adds sharp lib (production webp optimization - https://nextjs.org/docs/messages/sharp-missing-in-production) 
- [x] adds new thumbor URL as images domains https://nextjs.org/docs/basic-features/image-optimization#domains
- [x] uses next/future/image as default FS image component.
- [x] removes unnecessary styles.
- [x] tweaks Image Gallery Selectors ref and styles.

## How to test it?

- you can run `yarn build && yarn serve` locally and test the images from all the pages. Check Devtools tab, filter img and double-check the domain, type, size, and time for each image, along with the cache from the response headers.
(expected webp format and HIT the Cloudfront cache)
- you can use the starter PR and check if the images are working as expected webp format and HIT the Cloudfront cache in Devtols response Headers.

### Starters Deploy Preview

https://github.com/vtex-sites/starter.store/pull/18

## References

https://nextjs.org/docs/messages/sharp-missing-in-production
https://nextjs.org/docs/api-reference/next/image#loader
https://nextjs.org/docs/basic-features/image-optimization#loaders
https://nextjs.org/docs/api-reference/next/image#loader-configuration
https://nextjs.org/docs/basic-features/image-optimization#domains
https://nextjs.org/docs/api-reference/next/image#caching-behavior
https://help.vtex.com/en/announcements/na-vtex-conversao-de-png-para-webp-agora-e-lossless--4J4ZtrCZIksY8qikWgeomY

### CODE REVIEW

File: packages/core/src/components/ui/Image/thumborUrlBuilder.ts [hunk 1]
```diff
@@ -19,7 +21,7 @@ export interface ThumborOptions {
   manualCrop?: Box | false
 }
 
-const THUMBOR_SERVER = 'https://assets.vtex.app'
+const THUMBOR_SERVER = `https://${storeConfig.api.storeId}.vtexassets.com`
```
Comment (eduardoformiga) [inline packages/core/src/components/ui/Image/thumborUrlBuilder.ts:24]: new thumbor URL by account

File: packages/core/next.config.js [hunk 2]
```diff
@@ -1,10 +1,15 @@
 // @ts-check
+const storeConfig = require('./faststore.config')
 
 /**
  * @type {import('next').NextConfig}
  * */
 const nextConfig = {
   /* config options here */
+  images: {
+    domains: [`${storeConfig.api.storeId}.vtexassets.com`],
```
Comment (eduardoformiga) [inline packages/core/next.config.js:10]: new thumbor URL by account

File: packages/core/package.json [hunk 3]
```diff
@@ -51,6 +51,7 @@
     "react-intersection-observer": "^8.32.5",
     "sass": "^1.44.0",
     "sass-loader": "^12.6.0",
+    "sharp": "^0.31.3",
```
Comment (eduardoformiga) [inline packages/core/package.json:54]: webp for production environment.

File: packages/core/src/components/ui/Image/Image.tsx [hunk 4]
```diff
@@ -1,59 +1,34 @@
...
-  }
+// Next loader function does not handle all props as height and options,
+// so we use the useImage hook to handle the custom thumbor loader with VTEX CDN
+// https://nextjs.org/docs/api-reference/next/image#loader
+function Image({ src, width, height, quality, ...otherProps }: ImageProps) {
+  const { src: thumborSrc, alt } = useImage({
+    src: String(src),
+    width: Number(width),
+    height: Number(height),
+    options: quality ? ({ filters: { quality } } as ThumborOptions) : undefined,
```
Comment (eduardoformiga) [inline packages/core/src/components/ui/Image/Image.tsx:15]: easy way to handle quality

File: packages/core/src/components/ui/ImageGallery/image-gallery-selector.module.scss [hunk 5]
```diff
@@ -96,7 +96,7 @@
       color: var(--fs-color-text-display);
     }
 
-    span {
+    span[data-fs-icon] {
```
Comment (eduardoformiga) [inline packages/core/src/components/ui/ImageGallery/image-gallery-selector.module.scss:99]: Fixes this span style
Screenshot 2023-03-10 at 20 12 43

**Review by ArthurTriis1 — APPROVED** [review]
Awesome!

File: packages/core/src/components/ui/ImageGallery/image-gallery-selector.module.scss (hunk 5 above)
Comment (hellofanny) [reply packages/core/src/components/ui/ImageGallery/image-gallery-selector.module.scss:99]: can we just sort the order?

`[data-fs-icon]span`

File: packages/core/src/components/ui/ImageGallery/image-gallery-selector.module.scss [hunk 6]
```diff
@@ -124,7 +124,7 @@
       width: 100%;
       background: linear-gradient(180deg, var(--fs-color-body-bkg) 55%, transparent);
 
-      span {
+      span[data-fs-icon] {
```
Comment (hellofanny) [inline packages/core/src/components/ui/ImageGallery/image-gallery-selector.module.scss:127]:
```suggestion
      [data-fs-icon]span {
```

**Review by hellofanny — APPROVED** [review]
👏  WOW!
Excellent work formiga! I've tested it on the preview, and it works as expected.
Thank you very much for the reference links!!

File: packages/core/src/components/ui/Image/thumborUrlBuilder.ts [hunk 7]
```diff
@@ -1,3 +1,5 @@
+import storeConfig from '../../../../faststore.config'
```
[the remark this answers is not in the document]
Comment (eduardoformiga) [reply packages/core/src/components/ui/Image/thumborUrlBuilder.ts]: done [here](https://github.com/vtex/faststore/pull/1661/commits/89c2e9cac02e252a4c1cdb2fb7a1b1f0e3bc605c).

File: packages/core/src/components/ui/ImageGallery/image-gallery-selector.module.scss (hunk 6 above)
Comment (eduardoformiga) [reply packages/core/src/components/ui/ImageGallery/image-gallery-selector.module.scss]: done [here](https://github.com/vtex/faststore/pull/1661/commits/e35bb2bd40f0c380343446bf3d108bbf2f693c87).

**Pull request merged**